pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,324 @@
+
+        for i, file in enumerate(filenames):
+            if not overwrite and os.path.basename(file) in self._stored_files.keys():
+                self._logger.warning(f"{file} not copied, since already present and 'overwrite'=False.")
+                continue
+            _metadata = metadata[i] if isinstance(metadata, list) else metadata
+            self._store_file(file, _metadata)
+
+        self.run()
+
+    def _store_file(self, file, metadata):
+        try:
+            if self._storage_type.local:
+                shutil.copy(file, self.working_directory)
+            elif self._storage_type.s3:
+                self._external_storage.upload(file, metadata)
+        except Exception as e:
+            raise IOError(f"Storing {file} failed") from e
+        else:
+            self._stored_files[os.path.basename(file)] = metadata
+
+    def _remove_file(self, file):
+        try:
+            if self._storage_type.local:
+                os.remove(os.path.join(self.working_directory, file))
+            elif self._storage_type.s3:
+                self._external_storage.remove_file(file)
+        except Exception as e:
+            raise IOError(f"Removing {file} failed") from e
+        else:
+            del self._stored_files[file]
+
+    def remove_files(self, filenames, dryrun=True, raise_error=True, silent=False):
+        """Remove files in the storage
+
+        Parameters:
+            filenames(str/list of str):     Files to remove from the Storage
+            dryrun(bool):                   If true, only report the files to be removed (default=True)
+            raise_error(bool):              If true, missing files raise a FileNotFoundError (nothing gets removed),
+                                            else remove all files found.
+            silent(bool):                   If true, do not print the status message.
+        """
+        if isinstance(filenames, str):
+            filenames = [filenames]
+
+        files_not_found = [file for file in filenames if file not in self._stored_files]
+        if len(files_not_found) > 0 and raise_error:
+            raise FileNotFoundError(f"Files {files_not_found} not found, abort removing files. You may specify "
+                                    f"raise_error=False to suppress this error.")
+
+        files_to_remove = [file for file in filenames if file in self._stored_files]
+        if dryrun and not silent:
+            print(f"dryrun: to remove {files_to_remove} specify dryrun=False")
+            return
+
+        for file in files_to_remove:
+            self._remove_file(file)
+
+        if not silent:
+            print(f"removed {files_to_remove}")
+        self.run()
+
+    def run_static(self):
+        raise NotImplementedError('There is no static run mode for a Storage job.')
+
+    def collect_output(self):
+        pass
+
+    def append(self, job):
+        pass
```

Comment:
I would prefer if this raised an error. I guess it's a mandatory method from the parent class and appending jobs is meaningless here, but without an error I worry that people will pass a string and think they've just successfully appended a file to storage.